### ISSUE TITLE
Write the mutation function

### ISSUE THREAD

**Issue by alexdboxall** [mazfil/lab-allocator#51]
The `Schedule.mutate` function should be written. This should take a schedule, and make a **slight** change to it (e.g. move a single lab around to a different time, or swap two labs around), etc.

**Comment by Hexuan-Meng** [comment]
The mutate function is now implemented by moving a random lab to another time in the same room. The reason for moving the lab to the same room is to ensure that we have enough seats for that course. If we move a lab to another room with fewer seats, we may not fulfill the needs for all students.

I did not implement the function of swapping two labs, because a series of conditions needs to be considered before swapping them, such as if the two labs are of the same course, if they have different lengths, or if they are in different rooms. If we pick two labs randomly, we may not always have them met those conditions so that they can be swapped, especially when the schedule has little free space for allocation.

I will move onto other issues if the current implementation is deemed satisfactory. Feel free to share any thoughts or modify the implementation if you have better ideas.
